### PR TITLE
Fixed crash in `getCodeOfSubregion`

### LINKED ISSUE #1775: Python frontend crashes with `java.lang.StringIndexOutOfBoundsException`
# Stack trace

```
java.util.concurrent.ExecutionException: java.lang.StringIndexOutOfBoundsException: Range [47, 249) out of bounds for length 247
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at de.fraunhofer.aisec.cpg_vis_neo4j.Application.call(Application.kt:612)
	at de.fraunhofer.aisec.cpg_vis_neo4j.Application.call(Application.kt:88)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at de.fraunhofer.aisec.cpg_vis_neo4j.ApplicationKt.main(Application.kt:653)
Caused by: java.lang.StringIndexOutOfBoundsException: Range [47, 249) out of bounds for length 247
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:55)
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:52)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex(Preconditions.java:112)
	at java.base/jdk.internal.util.Preconditions.checkFromToIndex(Preconditions.java:349)
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4914)
	at java.base/java.lang.String.substring(String.java:2876)
	at de.fraunhofer.aisec.cpg.helpers.RegionUtilsKt.getCodeOfSubregion(RegionUtils.kt:72)
	at de.fraunhofer.aisec.cpg.graph.NodeBuilderKt.codeAndLocationFromChildren(NodeBuilder.kt:357)
	at de.fraunhofer.aisec.cpg.frontends.python.StatementHandler.makeBlock(StatementHandler.kt:781)
	at de.fraunhofer.aisec.cpg.frontends.python.StatementHandler.handleFunctionDef(StatementHandler.kt:531)
	at de.fraunhofer.aisec.cpg.frontends.python.StatementHandler.handleClassDef(StatementHandler.kt:436)
	at de.fraunhofer.aisec.cpg.frontends.python.StatementHandler.handleNode(StatementHandler.kt:59)
	at de.fraunhofer.aisec.cpg.frontends.python.StatementHandler.handleNode(StatementHandler.kt:54)
	at de.fraunhofer.aisec.cpg.frontends.python.PythonHandler.handle(PythonHandler.kt:43)
	at de.fraunhofer.aisec.cpg.frontends.python.PythonLanguageFrontend.pythonASTtoCPG(PythonLanguageFrontend.kt:270)
	at de.fraunhofer.aisec.cpg.frontends.python.PythonLanguageFrontend.parse(PythonLanguageFrontend.kt:79)
	at de.fraunhofer.aisec.cpg.TranslationManager.parse(TranslationManager.kt:387)
	at de.fraunhofer.aisec.cpg.TranslationManager.parseSequentially(TranslationManager.kt:340)
	at de.fraunhofer.aisec.cpg.TranslationManager.runFrontends(TranslationManager.kt:240)
	at de.fraunhofer.aisec.cpg.TranslationManager.analyzeNonAsync(TranslationManager.kt:88)
	at de.fraunhofer.aisec.cpg.TranslationManager.analyze$lambda$0(TranslationManager.kt:70)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1489)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2071)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2033)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
```

# Unexpected behavior

It crashes

# Code of interest

http://github.com/blakeblackshear/frigate

### PR DESCRIPTION
**Pull request by oxisto**
Fixes #1775

### CODE REVIEW

**Comment by KuechA** [comment]
Can we add a test case for this?

File: cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/RegionUtils.kt [hunk 1]
```diff
@@ -62,12 +63,17 @@ fun getCodeOfSubregion(code: String, nodeRegion: Region, subRegion: Region): Str
...
             subRegion.endColumn - nodeRegion.startColumn
         } else {
             (StringUtils.ordinalIndexOf(code, nlType, subRegion.endLine - nodeRegion.startLine) +
                 subRegion.endColumn)
         }
+
+    // Unfortunately, we sometimes have issues with (non)-Unicode characters in code, where the
+    // python AST thinks that multiple characters are needed and reports a position that is actually
+    // beyond our "end"
+    end = min(end, code.length)
```
Comment (konradweiss) [inline cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/RegionUtils.kt:77]: Does that mean that in cases where the end is not reached,  (non)-Unicode characters are causing wrong code to be extracted for the subregion?

Comment (oxisto) [reply cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/RegionUtils.kt:77]: That could very well be, but there is nothing to do about it without debugging the python parser.

**Comment by oxisto** [comment]
> Can we add a test case for this?

Done

**Review by KuechA — APPROVED** [review]

**Pull request merged**